pixeltable/catalog/catalog.py

```diff
@@ -854,15 +857,19 @@ def _get_dir_entry(self, dir_id: UUID, name: str, lock_entry: bool = False) -> O
         )
         tbl_id = conn.execute(q).scalar_one_or_none()
         if tbl_id is not None:
-            if tbl_id not in self._tbls:
-                _ = self._load_tbl(tbl_id)
-            return self._tbls[tbl_id]
+            if version is not None:
+                return self._load_tbl_with_specified_version(tbl_id, version)
```

Comment:
Let's expand _tbls to be able to cache these as well, no need to treat them differently (it's still a valid Table).